lib/service/storage/elasticsearch.js

```diff
@@ -822,6 +822,71 @@ class ElasticSearch extends Service {
     }
   }
 
+  /**
+   * Delete fields of a document and replace it
+   *
+   * @param {String} index - Index name
+   * @param {String} collection - Collection name
+   * @param {String} id - Document id
+   * @param {Array}  fields - Document fields to be removed
+   * @param {Object} options - refresh (undefined), userId (null)
+   *
+   * @returns {Promise.<{ _id, _version, _source }>}
+   */
+  async deleteFields (
+    index,
+    collection,
+    id,
+    fields,
+    { refresh, userId=null } = {})
+  {
+    const esIndex = this._getESIndex(index, collection);
+    const esRequest = {
+      id,
+      index: esIndex,
+    };
+
+    // metadata
+    const _kuzzle_info = {
+      author: getUserId(userId),
+      updatedAt: Date.now(),
+      updater: getUserId(userId)
+    };
+
+    try {
+      debug('DeleteFields document: %o', esRequest);
+      const { body } = await this._client.get(esRequest);
+
+      for (const field of fields) {
+        _.set(body._source, field, undefined);
+      }
+
+      body._source._kuzzle_info = Object.assign(
+        body._source._kuzzle_info || { createdAt: Date.now() },
+        _kuzzle_info);
```

Comment:
Prefer the new deconstruction syntax like:
```
const newObj = { ...oldObj, value: 42 };
```
But also here you don't need to put the current metadata in you request, ES will perform an update of the metadata you provide (`updater` and `updatedAt`) Look [here](https://github.com/kuzzleio/kuzzle/blob/master/lib/service/storage/elasticsearch.js#L601) for an example

Reply:
Does ES do an update of the metadata even if the whole document is replaced ?

Reply:
Oh yes you're right, I forgot that this was a replace. Then yes you need to pass the existing metadata object enriched with the `updater` and `updatedAt` properties